management/vmutils/integration_test.go

```diff
@@ -74,7 +74,22 @@ func TestDeployPlatformCaptureRedeploy(t *testing.T) {
 	}
 
 	t.Logf("Shutting down VM: %s", vmname)
-	vmc.ShutdownRole(vmname, vmname, vmname)
+	if err := Await(client, func() (management.OperationId, error) {
+		return vmc.ShutdownRole(vmname, vmname, vmname)
+	}); err != nil {
+		t.Error(err)
+	}
+
+	for {
+		deployment, err := vmc.GetDeployment(vmname, vmname)
+		if err != nil {
+			t.Fatal(err)
+		}
+		if deployment.RoleInstanceList[0].PowerState == vm.PowerStateStopped {
+			break
+		}
+		time.Sleep(2 * time.Second)
```

Comment:
Create helper WaitForRoleState. 

I used to have it another project checked in some internal sd repo and it helped a great deal. :-) actually you can make it per deployment and get all role states and compare against goal state:

WaitForRolesState(deploymentId, roleState).